### PR TITLE
hyperledger-fabric: init at 1.3.0

### LINKED ISSUE #52542: package request: hyperledger-fabric
Was not able to get it up & running, check https://stackoverflow.com/questions/53839759/hyperledger-fabric-nix-derivation

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
closes https://github.com/NixOS/nixpkgs/issues/52542

cc: @rsoeldner

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/misc/hyperledger-fabric/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "hyperledger-fabric-${version}";
```
Comment (worldofpeace) [inline pkgs/tools/misc/hyperledger-fabric/default.nix]:
```suggestion
  pname = "hyperledger-fabric";
```
`buildGoPackage` can have just a `pname` and a `version` now.

File: pkgs/tools/misc/hyperledger-fabric/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    rev = "v${version}";
+    sha256 = "08qrrxzgkqg9v7n3y8f2vggyqx9j65wisxi17hrabz5mzaq299xs";
+  };
+
+  meta = with stdenv.lib; {
+    description = "An implementation of blockchain technology, leveraging familiar and proven technologies";
+    homepage = https://wiki.hyperledger.org/projects/Fabric;
+    license = licenses.asl20;
+    maintainers = [ maintainers.marsam ];
+    platforms = platforms.all;
```
Comment (worldofpeace) [inline pkgs/tools/misc/hyperledger-fabric/default.nix]:
```suggestion

```

This is set automatically by `buildGoPackage`
https://github.com/NixOS/nixpkgs/blob/ca0c253a8066d32518e015e1befa27879fffcde0/pkgs/development/go-modules/generic/default.nix#L234

File: pkgs/tools/misc/hyperledger-fabric/default.nix (hunk 1 above)
Comment (marsam) [reply pkgs/tools/misc/hyperledger-fabric/default.nix]: I forgot about this, fixed. Thanks!

**Comment by worldofpeace** [comment]
Looks fine and builds locally, I can see the help menus. Though I won't be any help in testing the function ATM.

**Comment by rsoeldner** [comment]
@worldofpeace I'm using this nix derivation already and (at least until now) it works perfectly.

**Comment by worldofpeace** [comment]
> @worldofpeace I'm using this nix derivation already and (at least until now) it works perfectly.

Sounds good, I'll merge this :+1:

**Pull request merged**